File: python/j12.arguments.py
#!/usr/bin/env python
import sys

# illustrative variables 
LAST  = len(sys.argv) # index of last argument
FIRST = 1             # index of first argument

# utility variables
count = FIRST         # initialize counter

print "The arugments passed are:"
# use count style loop
while count < LAST:
   arg = sys.argv[count]  # get arg using index
   # print count and argument using count index
   print " item %d: %s" % (count,arg) 
   count += 1             # increment counter

File: ruby/j12.arguments.rb
#!/usr/bin/ruby
# illustrative variables
first     = 0  # set index of first element
min_count = 0  # set minimum number of elements required
# utility variables
count     = 0  # initialize counter

puts "The arugments passed are:"
# conditional loop to process elements using shift
while ARGV.length > min_count do          # fetches updated # of args as long 
 puts " item #{count+=1}: #{ARGV[first]}" # output result of first argument
 ARGV.shift                               # remove first element
end
